pool/pool.go

```diff
@@ -0,0 +1,73 @@
+package pool
+
+import (
+	"log"
+
+	"github.com/pisign/pisign-backend/types"
+)
+
+// Pool holds multiple apis and handles registration and deletion
+type Pool struct {
+	registerChan   chan types.API
+	unregisterChan chan types.API
+	Map            map[types.API]bool
+	name           string
+}
+
+// NewPool generates a new pool
+func NewPool() *Pool {
+	return &Pool{
+		registerChan:   make(chan types.API),
+		unregisterChan: make(chan types.API),
+		Map:            make(map[types.API]bool),
+		name:           "main",
+	}
+}
+
+// List turns map into list
+func (pool *Pool) List() []types.API {
+	keys := make([]types.API, len(pool.Map))
+	i := 0
+	for k := range pool.Map {
+		keys[i] = k
+		i++
+	}
+	return keys
+}
+
+// Save the current state of the pool
+func (pool *Pool) Save() {
+	list := pool.List()
+	layout := Layout{List: list, Name: pool.name}
+	err := SaveLayout(layout)
+	if err != nil {
+		panic("Error saving layout from pool!")
+	}
+}
+
+// Start main entry point of a pool
+func (pool *Pool) Start() {
+	for {
+		select {
+		case api := <-pool.registerChan:
+			pool.Map[api] = true
+			log.Printf("New API: %s\n", api.GetName())
+			log.Println("Size of Connection Pool: ", len(pool.Map))
+			pool.Save()
+		case api := <-pool.unregisterChan:
+			delete(pool.Map, api)
```

Comment:
Is this how we can delete a widget from the layout?

Reply:
Not exactly, this just removes the api from the `Pool` object. We haven't fully fleshed out the delete process from the front end